libkbfs/folder_branch_ops.go

```diff
@@ -3905,14 +3905,16 @@ func (fbo *folderBranchOps) applyMDUpdatesLocked(ctx context.Context,
 
 	// If there's anything in the journal, don't apply these MDs.
 	// Wait for CR to happen.
-	mergedRev, err := fbo.getJournalPredecessorRevision(ctx)
-	if err != nil {
-		return err
-	}
-	if mergedRev != MetadataRevisionUninitialized {
-		fbo.log.CDebugf(ctx,
-			"Ignoring fetched revisions while MDs are in journal")
-		return nil
+	if fbo.isMasterBranchLocked(lState) {
```

Comment:
Because if we call `getJournalPredecessorRevision` while we're staged, it returns an error, and we end up failing this whole call, which was leading to failing tests from calls to `SyncFromServerForTesting`). Local squashes make it somewhat nondeterminstic when we'll be in staged mode w.r.t the tests.